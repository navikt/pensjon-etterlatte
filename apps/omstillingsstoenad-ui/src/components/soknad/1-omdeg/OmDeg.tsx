import SoknadSteg from '../../../typer/SoknadSteg'
import { useTranslation } from 'react-i18next'
import InnloggetBruker from './InnloggetBruker'
import { SkjemaGruppe } from '../../felles/SkjemaGruppe'
import { FormProvider, useForm } from 'react-hook-form'
import { IValg } from '../../../typer/Spoersmaal'
import { useSoknadContext } from '../../../context/soknad/SoknadContext'
import { ISoeker } from '../../../typer/person'
import { ActionTypes } from '../../../context/soknad/soknad'
import { RHFInput, RHFKontonummerInput, RHFTelefonInput } from '../../felles/rhf/RHFInput'
import { RHFInlineRadio, RHFSpoersmaalRadio } from '../../felles/rhf/RHFRadio'
import Feilmeldinger from '../../felles/Feilmeldinger'
import { useBrukerContext } from '../../../context/bruker/BrukerContext'
import Navigasjon from '../../felles/Navigasjon'
import { Cell, Grid, Heading, RadioProps } from '@navikt/ds-react'
import { BankkontoType } from '../../../typer/utbetaling'
import UtenlandskBankInfo from './utenlandskBankInfo/UtenlandskBankInfo'
import HvorforSpoerVi from '../../felles/HvorforSpoerVi'
import { deepCopy } from '../../../utils/deepCopy'
import { RHFSelect } from '../../felles/rhf/RHFSelect'
import { useLand } from '../../../hooks/useLand'
import { SkjemaElement } from '../../felles/SkjemaElement'
import Bredde from '../../../typer/bredde'

const OmDeg: SoknadSteg = ({ neste }) => {
    const { t } = useTranslation()
    const { state, dispatch } = useSoknadContext()
    const brukerState = useBrukerContext().state
    const { land }: { land: any } = useLand()
    const lagre = (data: ISoeker) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG, payload: { ...deepCopy(data), erValidert: true } })
        neste!!()
    }

    const methods = useForm<ISoeker>({
        defaultValues: state.omDeg || {},
        shouldUnregister: true,
    })

    const {
        handleSubmit,
        watch,
        formState: { errors },
    } = methods

    const skalSjekkeFlyktningStatus = brukerState.foedselsaar!! < 1960

    const oppholderSegINorge = watch('oppholderSegINorge')
    const bankkontoType = watch('utbetalingsInformasjon.bankkontoType')

    return (
        <>
            <SkjemaElement>
                <Heading size={'medium'} className={'center'}>
                    {t('omDeg.tittel')}
                </Heading>
            </SkjemaElement>
            <InnloggetBruker />

            <FormProvider {...methods}>
                {/* TODO: Flytte dette til start eller eget steg? */}

                <form>
                    <SkjemaGruppe>
                        {!brukerState.adressebeskyttelse && !brukerState.adresse && (
                            <SkjemaGruppe>
                                <RHFInput name={'alternativAdresse'} label={t('omDeg.alternativAdresse')} />
                            </SkjemaGruppe>
                        )}

                        {!brukerState.telefonnummer && (
                            <SkjemaGruppe>
                                <Grid>
                                    <Cell xs={12} md={6} className={'kol'}>
                                        <RHFTelefonInput
                                            htmlSize={Bredde.S}
                                            name={'kontaktinfo.telefonnummer'}
                                            label={t('omDeg.kontaktinfo.telefonnummer')}
                                            valgfri={true}
                                        />
                                    </Cell>
                                </Grid>
                            </SkjemaGruppe>
                        )}
                    </SkjemaGruppe>

                    {/* 2.7 */}
                    {!brukerState.adressebeskyttelse && (
                        <SkjemaGruppe>
                            <SkjemaElement>
                                <RHFSpoersmaalRadio
                                    name={'oppholderSegINorge'}
                                    legend={t('omDeg.oppholderSegINorge')}
                                    description={
                                        <HvorforSpoerVi title="omDeg.oppholderSegINorge">
                                            {t('omDeg.oppholdHvorfor')}
                                        </HvorforSpoerVi>
                                    }
                                />
                            </SkjemaElement>

                            {oppholderSegINorge === IValg.JA && (
                                <SkjemaGruppe>
                                    <RHFKontonummerInput
                                        htmlSize={Bredde.S}
                                        name={'utbetalingsInformasjon.kontonummer'}
                                        label={t('omDeg.utbetalingsInformasjon.kontonummer')}
                                        placeholder={t('felles.elleveSiffer')}
                                        description={t('omDeg.utbetalingsInformasjon.informasjon')}
                                    />
                                </SkjemaGruppe>
                            )}

                            {oppholderSegINorge === IValg.NEI && (
                                <>
                                    <SkjemaGruppe>
                                        <RHFSelect
                                            className="kol-50"
                                            name={`oppholdsland`}
                                            label={t('omDeg.oppholdsland')}
                                            selectOptions={land}
                                        />
                                    </SkjemaGruppe>

                                    <SkjemaElement>
                                        <RHFInlineRadio
                                            name={'utbetalingsInformasjon.bankkontoType'}
                                            legend={t('omDeg.utbetalingsInformasjon.bankkontoType')}
                                        >
                                            {Object.values(BankkontoType).map((value) => {
                                                return { children: t(value), value } as RadioProps
                                            })}
                                        </RHFInlineRadio>
                                    </SkjemaElement>

                                    {bankkontoType === BankkontoType.norsk && (
                                        <RHFKontonummerInput
                                            htmlSize={Bredde.S}
                                            name={'utbetalingsInformasjon.kontonummer'}
                                            label={t('omDeg.utbetalingsInformasjon.kontonummer')}
                                            placeholder={t('felles.elleveSiffer')}
                                        />
                                    )}

                                    {bankkontoType === BankkontoType.utenlandsk && <UtenlandskBankInfo />}
                                </>
                            )}
                        </SkjemaGruppe>
                    )}

                    {skalSjekkeFlyktningStatus && (
                        <SkjemaGruppe>
                            <RHFSpoersmaalRadio
                                name={'flyktning'}
                                legend={t('omDeg.flyktning')}
                                description={
                                    <HvorforSpoerVi title="omDeg.flyktning">
                                        {t('omDeg.flyktningHvorfor')}
                                    </HvorforSpoerVi>
                                }
                            />
                        </SkjemaGruppe>
                    )}

                    <br />

                    <Feilmeldinger errors={errors} />

                    <Navigasjon neste={{ onClick: handleSubmit(lagre) }} />
                </form>
            </FormProvider>
        </>
    )
}

export default OmDeg
