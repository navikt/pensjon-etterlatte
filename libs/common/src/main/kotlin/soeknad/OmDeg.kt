package soeknad

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonValue
import no.nav.etterlatte.libs.common.soeknad.Valg

@JsonIgnoreProperties(ignoreUnknown = true)
data class OmDeg(
    val bostedsadresseBekreftet: Valg?,
    val alternativAdresse: String?,
    val kontaktinfo: Kontaktinfo?,
    val utbetalingsInformasjon: UtbetalingsInformasjon?,
    val flyktning: Valg?,
    val oppholderSegINorge: Valg?,
    val oppholdsland: String?,
    val medlemFolketrygdenUtland: Valg?,
    val nySivilstatus: NySivilstatus?
)

data class Kontaktinfo(
    val epost: String?,
    val telefonnummer: String?
)

data class UtbetalingsInformasjon(
    val kontonummer: String?,
    val bankkontoType: BankkontoType?,
    val utenlandskBankNavn: String?,
    val utenlandskBankAdresse: String?,
    val iban: String?,
    val swift: String?
)

enum class BankkontoType(@get:JsonValue val beskrivelse: String) {
    norsk("bankkontoType.norsk"),
    utenlandsk("bankkontoType.utenlandsk")
}
