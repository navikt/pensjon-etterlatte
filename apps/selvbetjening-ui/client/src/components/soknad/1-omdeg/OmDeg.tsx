import "./OmDeg.scss";
import SoknadSteg from "../../../typer/SoknadSteg";
import { useTranslation } from "react-i18next";
import InnloggetBruker from "./InnloggetBruker";
import { RadioProps, SkjemaGruppe } from "nav-frontend-skjema";
import { FormProvider, useForm } from "react-hook-form";
import { IValg } from "../../../typer/Spoersmaal";
import { useSoknadContext } from "../../../context/soknad/SoknadContext";
import { ISoeker } from "../../../typer/person";
import { ActionTypes } from "../../../context/soknad/soknad";
import { RHFInput, RHFKontonummerInput, RHFTelefonInput } from "../../felles/RHFInput";
import { RHFInlineRadio, RHFSpoersmaalRadio } from "../../felles/RHFRadio";
import Feilmeldinger from "../../felles/Feilmeldinger";
import { useBrukerContext } from "../../../context/bruker/BrukerContext";
import Navigasjon from "../../felles/Navigasjon";
import { emailMatcher } from "../../../utils/matchers";
import { Cell, Grid, Heading } from "@navikt/ds-react";
import { BankkontoType } from "../../../typer/utbetaling";
import UtenlandskBankInfo from "./utenlandskBankInfo/UtenlandskBankInfo";
import HvorforSpoerVi from "../../felles/HvorforSpoerVi";
import SkjemaGruppering from "../../felles/SkjemaGruppering";
import { deepCopy } from "../../../utils/deepCopy";
import { RHFSelect } from "../../felles/RHFSelect";
import { useLand } from "../../../hooks/useLand";

const OmDeg: SoknadSteg = ({ neste }) => {
    const { t } = useTranslation();
    const { state, dispatch } = useSoknadContext();
    const brukerState = useBrukerContext().state;
    const { land }: { land: any } = useLand();
    const lagre = (data: ISoeker) => {
        dispatch({ type: ActionTypes.OPPDATER_OM_DEG, payload: { ...deepCopy(data), erValidert: true } });
        neste!!();
    };

    const methods = useForm<ISoeker>({
        defaultValues: state.omDeg || {},
        shouldUnregister: true,
    });

    const {
        handleSubmit,
        watch,
        formState: { errors },
    } = methods;

    const skalSjekkeFlyktningStatus = brukerState.foedselsaar!! < 1960;

    const borPaaRegistrertAdresse = watch("bostedsadresseBekreftet");
    const oppholderSegINorge = watch("oppholderSegINorge");
    const bankkontoType = watch("utbetalingsInformasjon.bankkontoType");

    return (
        <>
            {/* Steg 2 */}
            <Heading size={"medium"} className={"center"}>
                {t("omDeg.tittel")}
            </Heading>

            {/* Informasjon om den innloggede brukeren */}
            <InnloggetBruker/>

            {/* Skjema for utfylling av info om innlogget bruker / s√∏ker */}
            <FormProvider {...methods}>
                {/* TODO: Flytte dette til start eller eget steg? */}

                <form>
                    <SkjemaGruppering>
                        {!brukerState.adressebeskyttelse && (
                            <>
                                <RHFSpoersmaalRadio
                                    name={"bostedsadresseBekreftet"}
                                    legend={t("omDeg.bostedsadresseBekreftet")}
                                />

                                {borPaaRegistrertAdresse === IValg.NEI && (
                                    <SkjemaGruppe>
                                        <RHFInput
                                            name={"alternativAdresse"}
                                            label={t("omDeg.alternativAdresse")}
                                        />
                                    </SkjemaGruppe>
                                )}
                            </>
                        )}

                        <SkjemaGruppe>
                            <Grid>
                                <Cell xs={12} md={3} className={"kol"}>
                                    <RHFTelefonInput
                                        bredde={"S"}
                                        name={"kontaktinfo.telefonnummer"}
                                        label={t("omDeg.kontaktinfo.telefonnummer")}
                                        placeholder={t("omDeg.kontaktinfo.telefonnummerPlaceholder")}
                                        valgfri={true}
                                    />
                                </Cell>

                                <Cell xs={12} md={6} className={"kol"}>
                                    <RHFInput
                                        name={"kontaktinfo.epost"}
                                        label={t("omDeg.kontaktinfo.epost")}
                                        rules={{ pattern: emailMatcher }}
                                    />
                                </Cell>
                            </Grid>
                        </SkjemaGruppe>
                    </SkjemaGruppering>

                    {/* 2.7 */}
                    {!brukerState.adressebeskyttelse && (
                        <SkjemaGruppering>
                            <RHFSpoersmaalRadio
                                name={"oppholderSegINorge"}
                                legend={t("omDeg.oppholderSegINorge")}
                                description={
                                    <HvorforSpoerVi title="omDeg.oppholderSegINorge">
                                        {t("omDeg.oppholdHvorfor")}
                                    </HvorforSpoerVi>
                                }
                            />

                            {oppholderSegINorge === IValg.JA && (
                                <SkjemaGruppe>
                                    <RHFKontonummerInput
                                        bredde={"S"}
                                        name={"utbetalingsInformasjon.kontonummer"}
                                        label={t("omDeg.utbetalingsInformasjon.kontonummer")}
                                        placeholder={"11 siffer"}
                                        description={t("omDeg.utbetalingsInformasjon.informasjon")}
                                    />
                                </SkjemaGruppe>
                            )}

                            {oppholderSegINorge === IValg.NEI && (
                                <>
                                    <SkjemaGruppe>
                                        <RHFSelect
                                            className="kol-50"
                                            name={`oppholdsland`}
                                            label={t("omDeg.oppholdsland")}
                                            selectOptions={land}
                                        />
                                    </SkjemaGruppe>

                                    <RHFSpoersmaalRadio
                                        name={"medlemFolketrygdenUtland"}
                                        legend={t("omDeg.medlemFolketrygdenUtland")}
                                    />

                                    <RHFInlineRadio
                                        name={"utbetalingsInformasjon.bankkontoType"}
                                        legend={t("omDeg.utbetalingsInformasjon.bankkontoType")}
                                        radios={Object.values(BankkontoType).map((value) => {
                                            return { label: t(value), value } as RadioProps;
                                        })}
                                    />

                                    {bankkontoType === BankkontoType.norsk && (
                                        <RHFKontonummerInput
                                            bredde={"S"}
                                            name={"utbetalingsInformasjon.kontonummer"}
                                            label={t("omDeg.utbetalingsInformasjon.kontonummer")}
                                            placeholder={"11 siffer"}
                                        />
                                    )}

                                    {bankkontoType === BankkontoType.utenlandsk && <UtenlandskBankInfo/>}
                                </>
                            )}
                        </SkjemaGruppering>
                    )}

                    {skalSjekkeFlyktningStatus && (
                        <SkjemaGruppe>
                            <RHFSpoersmaalRadio
                                name={"flyktning"}
                                legend={t("omDeg.flyktning")}
                                description={
                                    <HvorforSpoerVi title="omDeg.flyktning">
                                        {t("omDeg.flyktningHvorfor")}
                                    </HvorforSpoerVi>
                                }
                            />
                        </SkjemaGruppe>
                    )}

                    <br/>

                    <Feilmeldinger errors={errors}/>

                    <Navigasjon neste={{ onClick: handleSubmit(lagre) }}/>
                </form>
            </FormProvider>
        </>
    );
};

export default OmDeg;
