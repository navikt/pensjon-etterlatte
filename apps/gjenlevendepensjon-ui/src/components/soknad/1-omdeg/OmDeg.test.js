import { render, fireEvent } from "@testing-library/react";
import * as JSutils from "nav-frontend-js-utils";
import * as uuid from "uuid";
import OmDeg from "./OmDeg";

jest.mock("react-i18next", () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    ...jest.requireActual("react-i18next"),
    useTranslation: () => {
        return {
            t: (str) => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
}));

// Her må det mockes litt forskjellige guid-pakker...
jest.mock("uuid", () => ({
    v4: () => "456"
}))
describe("Om deg og avdød", () => {
    it("Snapshot", () => {
        jest.spyOn(JSutils, "guid").mockReturnValue("123");

        const { container, getByText } = render(<OmDeg />);
        
        const oppholderSegINorge = container.querySelectorAll("[name='oppholderSegINorge'][value='radiobuttons.nei']")[0];
        fireEvent.click(oppholderSegINorge);
        const bankkontoTypeUtenlandsk = container.querySelectorAll("[value='bankkontoType.utenlandsk']")[0];
        fireEvent.click(bankkontoTypeUtenlandsk);

        expect(getByText("omDeg.utbetalingsInformasjon.iban")).toBeDefined();
        expect(container).toMatchSnapshot();
    });
});
