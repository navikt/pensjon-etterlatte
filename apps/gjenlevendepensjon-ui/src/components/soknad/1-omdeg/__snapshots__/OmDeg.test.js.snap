// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Om deg og avd√∏d Snapshot 1`] = `
<div>
  <div
    class="sc-hLseeU jLGscI"
  >
    <h1
      class="center navds-heading navds-heading--medium"
    >
      omDeg.tittel
    </h1>
  </div>
  <form>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <fieldset
        class="sc-beqWaB kWPIyU"
      >
        <div>
          <div
            class="navds-form-field navds-form-field--medium"
          >
            <label
              class="navds-form-field__label navds-label"
              for="alternativAdresse"
            >
              omDeg.alternativAdresse
            </label>
            <input
              aria-invalid="false"
              class="navds-text-field__input navds-body-short navds-body-medium"
              id="alternativAdresse"
              required=""
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-form-field__error"
              id="textField-error-r0"
            />
          </div>
        </div>
      </fieldset>
      <fieldset
        class="sc-beqWaB kWPIyU"
      >
        <div
          class="navds-grid"
        >
          <div
            class="navds-grid__cell navds-grid__cell--xs-12 navds-grid__cell--md-6 kol"
          >
            <div
              class="navds-form-field navds-form-field--medium"
            >
              <label
                class="navds-form-field__label navds-label"
                for="kontaktinfo.telefonnummer"
              >
                omDeg.kontaktinfo.telefonnummer
              </label>
              <input
                aria-invalid="false"
                class="navds-text-field__input navds-body-short navds-body-medium"
                id="kontaktinfo.telefonnummer"
                required=""
                size="25"
                type="text"
                value=""
              />
              <div
                aria-live="polite"
                aria-relevant="additions removals"
                class="navds-form-field__error"
                id="textField-error-r1"
              />
            </div>
          </div>
        </div>
      </fieldset>
    </fieldset>
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <div
        class="sc-hLseeU jLGscI"
      >
        <div
          id="oppholderSegINorge"
        >
          <fieldset
            aria-invalid="false"
            class="sc-eDDNvR etxXkD navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
          >
            <legend
              class="navds-fieldset__legend navds-label"
            >
              omDeg.oppholderSegINorge
            </legend>
            <div
              class="navds-fieldset__description navds-body-long navds-body-long--small"
              id="fieldset-description-r3"
            >
              <div>
                <button
                  aria-expanded="false"
                  class="sc-hAtEyd gExcfY"
                  data-testid="hvorforPanel_toggle"
                  type="button"
                >
                  <span>
                    hvorforSpoerVi
                  </span>
                  <span>
                    <svg
                      fill="none"
                      focusable="false"
                      height="1em"
                      role="img"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M12 17.414 4.293 9.707l1.414-1.414L12 14.586l6.293-6.293 1.414 1.414L12 17.414Z"
                        fill="currentColor"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
            <div
              class="navds-radio-buttons"
            >
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  aria-invalid="false"
                  class="navds-radio__input"
                  id="radio-r5"
                  name="oppholderSegINorge"
                  type="radio"
                  value="radiobuttons.ja"
                />
                <label
                  class="navds-radio__label"
                  for="radio-r5"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short"
                    >
                      radiobuttons.ja
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  aria-invalid="false"
                  class="navds-radio__input"
                  id="radio-r6"
                  name="oppholderSegINorge"
                  type="radio"
                  value="radiobuttons.nei"
                />
                <label
                  class="navds-radio__label"
                  for="radio-r6"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short"
                    >
                      radiobuttons.nei
                    </span>
                  </span>
                </label>
              </div>
            </div>
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-fieldset__error"
              id="fieldset-error-r3"
            />
          </fieldset>
        </div>
      </div>
      <fieldset
        class="sc-beqWaB kWPIyU"
      >
        <div
          id="oppholdsland"
        >
          <div
            class="kol-50 navds-form-field navds-form-field--medium"
          >
            <label
              class="navds-form-field__label navds-label"
              for="textField-r7"
            >
              omDeg.oppholdsland
            </label>
            <div
              class="navds-select__container"
            >
              <select
                aria-invalid="false"
                class="navds-select__input navds-body-short navds-body--medium"
                id="textField-r7"
              >
                <option
                  value=""
                >
                  felles.velgLand
                </option>
              </select>
              <svg
                aria-hidden="true"
                class="navds-select__chevron"
                fill="none"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M12 17.414 4.293 9.707l1.414-1.414L12 14.586l6.293-6.293 1.414 1.414L12 17.414Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-form-field__error"
              id="textField-error-r7"
            />
          </div>
        </div>
      </fieldset>
      <div
        class="sc-hLseeU jLGscI"
      >
        <div
          id="utbetalingsInformasjon.bankkontoType"
        >
          <fieldset
            aria-invalid="false"
            class="sc-eDDNvR etxXkD navds-radio-group navds-radio-group--medium navds-fieldset navds-fieldset--medium"
          >
            <legend
              class="navds-fieldset__legend navds-label"
            >
              omDeg.utbetalingsInformasjon.bankkontoType
            </legend>
            <div
              class="navds-radio-buttons"
            >
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  aria-invalid="false"
                  class="navds-radio__input"
                  id="radio-rb"
                  name="utbetalingsInformasjon.bankkontoType"
                  type="radio"
                  value="bankkontoType.norsk"
                />
                <label
                  class="navds-radio__label"
                  for="radio-rb"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short"
                    >
                      bankkontoType.norsk
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="radioBorder navds-radio navds-radio--medium"
              >
                <input
                  aria-invalid="false"
                  class="navds-radio__input"
                  id="radio-rc"
                  name="utbetalingsInformasjon.bankkontoType"
                  type="radio"
                  value="bankkontoType.utenlandsk"
                />
                <label
                  class="navds-radio__label"
                  for="radio-rc"
                >
                  <span
                    class="navds-radio__content"
                  >
                    <span
                      class="navds-body-short"
                    >
                      bankkontoType.utenlandsk
                    </span>
                  </span>
                </label>
              </div>
            </div>
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-fieldset__error"
              id="fieldset-error-ra"
            />
          </fieldset>
        </div>
      </div>
      <div
        class="sc-hLseeU jLGscI"
      >
        <h1
          class="navds-heading navds-heading--small"
        >
          omDeg.utbetalingsInformasjon.tittel
        </h1>
      </div>
      <div
        class="sc-hLseeU jLGscI"
      >
        <div>
          <div
            class="navds-form-field navds-form-field--medium"
          >
            <label
              class="navds-form-field__label navds-label"
              for="utbetalingsInformasjon.utenlandskBankNavn"
            >
              omDeg.utbetalingsInformasjon.utenlandskBankNavn
            </label>
            <input
              aria-invalid="false"
              class="navds-text-field__input navds-body-short navds-body-medium"
              id="utbetalingsInformasjon.utenlandskBankNavn"
              required=""
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-form-field__error"
              id="textField-error-rd"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-hLseeU jLGscI"
      >
        <div>
          <div
            class="navds-form-field navds-form-field--medium"
          >
            <label
              class="navds-form-field__label navds-label"
              for="utbetalingsInformasjon.utenlandskBankAdresse"
            >
              omDeg.utbetalingsInformasjon.utenlandskBankAdresse
            </label>
            <input
              aria-invalid="false"
              class="navds-text-field__input navds-body-short navds-body-medium"
              id="utbetalingsInformasjon.utenlandskBankAdresse"
              required=""
              type="text"
              value=""
            />
            <div
              aria-live="polite"
              aria-relevant="additions removals"
              class="navds-form-field__error"
              id="textField-error-re"
            />
          </div>
        </div>
      </div>
      <div
        class="sc-hLseeU jLGscI"
      >
        <div
          class="navds-form-field navds-form-field--medium"
        >
          <label
            class="navds-form-field__label navds-label"
            for="utbetalingsInformasjon.iban"
          >
            <div
              class="sc-ktEKTO DTLKX"
            >
              omDeg.utbetalingsInformasjon.iban
              ¬†
              <div
                class="navds-help-text"
              >
                <button
                  aria-expanded="false"
                  class="navds-help-text__button"
                  type="button"
                >
                  <svg
                    aria-labelledby="title-rg"
                    class="navds-help-text__icon"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-rg"
                    >
                      hjelp
                    </title>
                    <path
                      clip-rule="evenodd"
                      d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0Zm0 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2Zm0 14a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Zm0-11c2.205 0 4 1.657 4 3.693 0 .986-.416 1.914-1.172 2.612l-.593.54-.294.28c-.477.466-.869.94-.936 1.417l-.01.144v.814h-1.991v-.814c0-1.254.84-2.214 1.675-3.002l.74-.68c.38-.35.59-.816.59-1.31 0-1.024-.901-1.856-2.01-1.856-1.054 0-1.922.755-2.002 1.71l-.006.145H8C8 6.657 9.794 5 12 5Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <svg
                    aria-labelledby="title-rh"
                    class="navds-help-text__icon navds-help-text__icon--filled"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-rh"
                    >
                      hjelp
                    </title>
                    <path
                      clip-rule="evenodd"
                      d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0Zm0 16a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Zm0-11c2.205 0 4 1.657 4 3.693 0 .986-.416 1.914-1.172 2.612l-.593.54-.294.28c-.477.466-.869.94-.936 1.417l-.01.144v.814h-1.991v-.814c0-1.254.84-2.214 1.675-3.002l.74-.68c.38-.35.59-.816.59-1.31 0-1.024-.901-1.856-2.01-1.856-1.054 0-1.922.755-2.002 1.71l-.006.145H8C8 6.657 9.794 5 12 5Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
                <div
                  aria-hidden="true"
                  class="navds-popover navds-help-text__popover navds-popover--hidden"
                  data-placement="top"
                  style="position: absolute; top: 0px; left: 0px;"
                  tabindex="-1"
                >
                  <div
                    class="navds-popover__content navds-body-short"
                  >
                    omDeg.utbetalingsInformasjon.ibanHjelpetekst
                  </div>
                  <div
                    class="navds-popover__arrow"
                    style="bottom: -0.5rem;"
                  />
                </div>
              </div>
            </div>
          </label>
          <input
            aria-invalid="false"
            class="navds-text-field__input navds-body-short navds-body-medium"
            id="utbetalingsInformasjon.iban"
            required=""
            size="40"
            type="text"
            value=""
          />
          <div
            aria-live="polite"
            aria-relevant="additions removals"
            class="navds-form-field__error"
            id="textField-error-rf"
          />
        </div>
      </div>
      <div
        class="sc-hLseeU jLGscI"
      >
        <div
          class="navds-form-field navds-form-field--medium"
        >
          <label
            class="navds-form-field__label navds-label"
            for="utbetalingsInformasjon.swift"
          >
            <div
              class="sc-ktEKTO DTLKX"
            >
              omDeg.utbetalingsInformasjon.swift
              ¬†
              <div
                class="navds-help-text"
              >
                <button
                  aria-expanded="false"
                  class="navds-help-text__button"
                  type="button"
                >
                  <svg
                    aria-labelledby="title-rj"
                    class="navds-help-text__icon"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-rj"
                    >
                      hjelp
                    </title>
                    <path
                      clip-rule="evenodd"
                      d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0Zm0 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10S17.523 2 12 2Zm0 14a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Zm0-11c2.205 0 4 1.657 4 3.693 0 .986-.416 1.914-1.172 2.612l-.593.54-.294.28c-.477.466-.869.94-.936 1.417l-.01.144v.814h-1.991v-.814c0-1.254.84-2.214 1.675-3.002l.74-.68c.38-.35.59-.816.59-1.31 0-1.024-.901-1.856-2.01-1.856-1.054 0-1.922.755-2.002 1.71l-.006.145H8C8 6.657 9.794 5 12 5Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <svg
                    aria-labelledby="title-rk"
                    class="navds-help-text__icon navds-help-text__icon--filled"
                    fill="none"
                    focusable="false"
                    height="1em"
                    role="img"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <title
                      id="title-rk"
                    >
                      hjelp
                    </title>
                    <path
                      clip-rule="evenodd"
                      d="M12 0c6.627 0 12 5.373 12 12s-5.373 12-12 12S0 18.627 0 12 5.373 0 12 0Zm0 16a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Zm0-11c2.205 0 4 1.657 4 3.693 0 .986-.416 1.914-1.172 2.612l-.593.54-.294.28c-.477.466-.869.94-.936 1.417l-.01.144v.814h-1.991v-.814c0-1.254.84-2.214 1.675-3.002l.74-.68c.38-.35.59-.816.59-1.31 0-1.024-.901-1.856-2.01-1.856-1.054 0-1.922.755-2.002 1.71l-.006.145H8C8 6.657 9.794 5 12 5Z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </button>
                <div
                  aria-hidden="true"
                  class="navds-popover navds-help-text__popover navds-popover--hidden"
                  data-placement="top"
                  style="position: absolute; top: 0px; left: 0px;"
                  tabindex="-1"
                >
                  <div
                    class="navds-popover__content navds-body-short"
                  >
                    omDeg.utbetalingsInformasjon.swiftHjelpetekst
                  </div>
                  <div
                    class="navds-popover__arrow"
                    style="bottom: -0.5rem;"
                  />
                </div>
              </div>
            </div>
          </label>
          <input
            aria-invalid="false"
            class="navds-text-field__input navds-body-short navds-body-medium"
            id="utbetalingsInformasjon.swift"
            required=""
            size="25"
            type="text"
            value=""
          />
          <div
            aria-live="polite"
            aria-relevant="additions removals"
            class="navds-form-field__error"
            id="textField-error-ri"
          />
        </div>
      </div>
    </fieldset>
    <br />
    <fieldset
      class="sc-beqWaB kWPIyU"
    >
      <div
        class="sc-jTrPJq jHZtnl undefined"
      >
        <button
          class="navds-button navds-button--primary navds-button--medium"
          type="button"
        >
          <span
            aria-live="polite"
            class="navds-label"
          >
            knapp.neste
          </span>
        </button>
      </div>
    </fieldset>
    <div
      class="sc-jTrPJq sc-gLDzan jHZtnl kSyGVc"
    >
      <button
        class="navds-button navds-button--secondary navds-button--medium"
        id="avbryt-btn"
        type="button"
      >
        <span
          aria-live="polite"
          class="navds-label"
        >
          knapp.avbryt
        </span>
      </button>
    </div>
  </form>
</div>
`;
